### DISCUSSION TITLE
Split Star Rating UI representation into 3 parts

### DISCUSSION THREAD

**Discussion by Givikap120** [ppy/osu#25600] [Ideas]
Considering that in future changes star rating most likely will merge with all multipliers like length bonus - we should make a better representation of the map difficulty in UI.

1) Star Rating - just a visual value to represent how difficult are diff spikes in the map are, kinda similar to the current star rating.
2) pp Rating - the difficulty to FC the map, with length bonus taken into account. Used later in the pp calculation.
3) Skills distribution - the pie chart is divided into 3 parts: Aim, Speed, and Reading.
- Aim will have a green to blue color depending on the amount of flow aim and snap aim in the map
- Speed will be just red
- Reading will be white to pink (same principle as in aim), where white is low AR (including overlap) + high AR readings, and pink will represent slider, rhythm (and other missed)

The color changing of skills can be confusing - but this info is useless for new players, while experienced players will get what changing color means by switching between jump and stream maps for example.

**Comment by peppy** [comment]
I think you should probably start by looking how star rating is implemented internally. Skill proportions already exist.

**Reply by Givikap120** [reply]
I've already made several rework proposals so I know how star rating works.
I'm talking about displaying it in the game.

**Reply by asdsadv** [reply]
Currently on stable, it shows both the local star rating and the actual star rating. As a player, I often look at both of these star ratings to determine what kind of map it most likely is (e.g if local star rating is lower than actual star rating, it is most likely a jump map. If the actual star rating is much higher than local, it's most likely a difficult slider map). While it's not a perfect method, it does provide some useful information. 
I think point 3 in this proposal or something similar that indicates the skillsets of a map would be useful (and better than the method above that can be done in stable), letting the player gauge what kind of map it is without playing, so they can tell if the map is in a skillset that they can play at that star rating.

**Reply by rodrigopina360** [reply]
Personally it would be cool to have a "jump map" tag or "stream map" tag, but it would be complicated to say what a map really is, especially as those specific skills are very subjective. A good example is Cry Thunder, I would say it's an aim map, because it features complicated aim sections, and it's present on most of the map. Someone else would say it's a stream map because it has a diff spike that features streams. In that example which would the correct tag be?
Besides that, I do slightly agree that there should be a way to know if a 6* map is consistently 6*, or if it's composed mainly by a 4* pattern, and then has a hard pattern that brings it up to 6*, but I also do agree that information like that would be bloating the UI even more.
